src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/GCHandle.T.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading;
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// Represents an strongly-typed opaque, GC handle to a managed object.
+    /// A GC handle is used when an object reference must be reachable from
+    /// unmanaged memory.
+    /// </summary>
+    /// <remarks>
+    /// <see cref="GCHandle{T}"/> corresponds to Normal roots.
+    /// For Weak and WeakTrackResurrection, see <see cref="WeakGCHandle{T}"/>.
+    /// For Pinned, see <see cref="PinnedGCHandle{T}"/>.
+    /// </remarks>
+    /// <seealso cref="GCHandle" />
+    /// <typeparam name="T">The type of the object this <see cref="GCHandle{T}"/> tracks to.</typeparam>
+    public struct GCHandle<T> : IDisposable
+        where T : class?
+    {
+        // The actual integer handle value that the EE uses internally.
+        private IntPtr _handle;
+
+        /// <summary>
+        /// Allocates a handle for the specified object.
+        /// </summary>
+        /// <param name="target">The object that uses the <see cref="GCHandle{T}"/>.</param>
+        public GCHandle(T target)
+        {
+            _handle = GCHandle.InternalAlloc(target, GCHandleType.Normal);
+        }
+
+        private GCHandle(IntPtr handle) => _handle = handle;
+
+        /// <summary>Determine whether this handle has been allocated or not.</summary>
+        public readonly bool IsAllocated => _handle != IntPtr.Zero;
+
+        /// <summary>Gets or sets the object this handle represents.</summary>
+        public readonly T Target
+        {
+            get
+            {
+                IntPtr handle = _handle;
+                GCHandle.ThrowIfInvalid(_handle);
+
+                return (T)GCHandle.InternalGet(handle)!;
```

Comment:
```suggestion
                return Unsafe.As<T>(GCHandle.InternalGet(_handle));
```

Unless things changed, I was under the impression that these new types would skip a bunch of safety checks.
@Sergio0694?

Reply:
Yes, the types should have maximum unsafety.

Reply:
I've left a few comments just now for this 👍